lib/web/server.cpp

```diff
@@ -314,7 +321,8 @@ string Socialite::Web::Server::cgiLaunch(string file, HttpHeader* header, string
 
 		while((error = ioctl(pipeFds[0], FIONREAD, &bytesAvailable)) == 0 && (waitOnId = waitpid(pid, &status, WNOHANG|WUNTRACED)) != pid) {
 			if(bytesAvailable > 0) {
-				buff = new char[bytesAvailable];
+				buff = new char[bytesAvailable + 1];
```

Comment:
Somewhat unrelated but this was causing intermittent problems. You need the extra `+1` buffer so there's an `\0` at the end otherwise you can get bad behavior.

Reply:
Repro: A route was server matched, but the server routes said it wasn't matched, so then it defaulted to a cgi call for the 404 page. And this was crashing the server.